go/saltpack/encrypt.go

```diff
@@ -149,17 +150,19 @@ func (es *encryptStream) init(sender BoxSecretKey, receivers []BoxPublicKey) err
 
 	eh.SenderSecretbox = secretbox.Seal([]byte{}, sender.GetPublicKey().ToKID(), (*[24]byte)(nonceForSenderKeySecretBox()), (*[32]byte)(&es.payloadKey))
 
-	for _, receiver := range receivers {
-		payloadKeyBox := ephemeralKey.Box(receiver, nonceForPayloadKeyBox(), es.payloadKey[:])
+	if !es.anonymous {
+		for _, receiver := range receivers {
+			payloadKeyBox := ephemeralKey.Box(receiver, nonceForPayloadKeyBox(), es.payloadKey[:])
 
-		keys := receiverKeys{PayloadKeyBox: payloadKeyBox}
+			keys := receiverKeys{PayloadKeyBox: payloadKeyBox}
 
-		// Don't specify the receivers if this public key wants to hide
-		if !receiver.HideIdentity() {
-			keys.ReceiverKID = receiver.ToKID()
-		}
+			// Don't specify the receivers if this public key wants to hide
+			if !receiver.HideIdentity() {
+				keys.ReceiverKID = receiver.ToKID()
```

Comment:
@jacobhaven note there already is support in the `go/saltpack` library for anonymous receivers via this mechanism.  The current API allows some receivers to be anonymous and some to be explicitly named.  I'm not saying we need to expose that all the way through to the CLI, but it might be at least useful to use this mechanism that's already here.